src/main/java/io/github/pascalklassen/pokefuture/PokemonService.java

```diff
@@ -111,6 +230,31 @@ private PokemonService() {
         return promise.future();
     }
 
+    public static void disableCashing(){
+        isCache = false;
+    }
+
+    public static void enableCashing(){
+        isCache = true;
+    }
+
+    @SuppressWarnings("unchecked")
+    private static <T> T getValue(@NotNull Class<T> clazz, @NotNull String uri){
+        Preconditions.checkNotNull(clazz);
+        Preconditions.checkNotNull(uri);
+        if(caches.containsKey(clazz)){
+             return (T) caches.get(clazz).getIfPresent(uri);
```

Comment:
Hierzu was allgemeines. Nicht so wichtig aber wäre cool wenn du das auch änderst.
Die Methode `getValue` sagt uns ja, dass sie uns einen Wert liefert. Aber der ControlFlow ließt sich so:
Wenn der Cache den Key beinhaltet, gib uns den Wert zurück, ansonsten Null. Das `ansonsten` sollte aber der Eigentliche Wert sein. Also wäre es viel lesbarer es so zu schreiben.

```java
// Das hier ist ein Sonderfall
if (!caches.containsKey(clazz) {
    return null;
}

// Am Ende steht der Normalfall wenn nichts schief geht
return (T) caches
            .get(clazz)
            .getIfPresent(uri);
```

Das ganze hat auch einen Namen: [Early out ODER Early Return Pattern](https://medium.com/swlh/return-early-pattern-3d18a41bba8)

### UND NOCH EINMAL
**WEHE DU KOPIERST DIESE SACHEN IN DEN EDITOR REIN, SCHREIB DAS GEFÄLLIGST!**